Added re subscription functionality

cutandpastey opened the pull request:
- Pathched live collection to allow for injection of models
- when injected models change trigger resubscribe
- added `resubscribe` method to cancel a subscription and re subscribe
  with required snapshot

cutandpastey committed:
Added re subscription functionality
  - Pathched live collection to allow for injection of models
  - when injected models change trigger resubscribe
  - added `resubscribe` method to cancel a subscription and re subscribe
    with required snapshot

cutandpastey committed:
tidy

trevorah commented:
looks ok to me, but i dont really know this library.

cutandpastey commented on lib/live-collection.js:
```diff
@@ -255,6 +254,18 @@ module.exports = Backbone.Collection.extend({
+    this.reset();
+    this.subscription.cancel();
+    this.subscription = null;
+    this.listen({ snapshot: true });
```
@suprememoocow should this be `this.snapshot`?

suprememoocow replied:
Not sure I quite follow the question @cutandpastey ?

suprememoocow replied:
The only thing I'm worried about here is that the original listen options will not be preserved. So, for example, if you chose not to use snapshots in the original subscribe, when the model change happens, a snapshot will be pulled...

cutandpastey replied:
Is it always the case when we resubscribe that we want a snapshot? Are some collections instantiated with `snapshot: false` for a specific reason? If so should we obey that here instead of just passing `true`?

cutandpastey replied:
@suprememoocow I think we are talking about the same thing so I will change it.

suprememoocow replied:
Additionally, it would be nice to merge the resolveChannel functionality (regarding the user) and the model into a single thing. This is probably overkill and not exactly required, but might make things simplier in the long run. 
Imagine, you had channel: `/users/:userId/rooms/:roomId` AND model `{ roomId: x, userId: y }`. Then, you just need to supply an object that had both those properties and we could make the resolveChannel functionality simpler... 

Instead of doing the stuff it does at the moment, it could simply look for all the `:param` elements and bind to changes to those elements on the model automatically. If any of them resolve to null, we don't subscribe until they're all there.

As I said, this is probably overkill, but it would make the `resolveChannel` functionality simpler...

cutandpastey replied:
Hmmm, I'm keen to get this change in as its relied upon in the webapp. Maybe file an issue and I can get onto it when the required feature have been merged in?

suprememoocow replied:
fine by me

cutandpastey committed:
update package.json version in preperation for a release

cutandpastey committed:
Merge branch 'master' into jp-resubscribe

cutandpastey committed:
update resubscribe function to acount for original collection settings

cutandpastey committed:
remove this.snapshot on resubscribe

cutandpastey committed:
update event binding

cutandpastey committed:
add comment

suprememoocow committed:
URL templates for live collections

suprememoocow committed:
faster travis builds

suprememoocow committed:
Added backbone url resolver

suprememoocow committed:
Added template subscription. Allows subscribers to subscribe to urlTemplates
When the context model changes, the template subscription is automatically
unsubscribed and resubscribed.

suprememoocow committed:
0.3.0

suprememoocow committed:
Allow TemplateSubscription to provide subscribe options

suprememoocow committed:
0.3.1

suprememoocow committed:
Reset live-collection before subscribe instead of on-snapshot, although
both might make more sense. The reason for before is that inconsistencies
can occur when chats for one room are displayed but the context is for
another room.

suprememoocow committed:
0.3.2

cutandpastey commented:
:+1:
